backbone/src/main/java/org/researchstack/backbone/ui/callbacks/StepViewCallback.java

```diff
@@ -0,0 +1,19 @@
+package org.researchstack.backbone.ui.callbacks;
+
+import org.researchstack.backbone.step.Step;
+import org.researchstack.backbone.ui.ViewTaskActivity;
+
+/**
+ * Interface used for callbacks related to changes in the Step.
+ * Used by {@link ViewTaskActivity} to communicate when a new step has been visualised
+ */
+public interface StepViewCallback {
+
+    /**
+     * This function is called when the {@link org.researchstack.backbone.ui.step.layout.StepLayout}
+     * of a @{@link Step} has been visualised and all animations have finished.
+     * @param activity the instance of the {@link ViewTaskActivity}
+     * @param step the @{@link Step} currently being shown
+     */
+    public void onStepShown(ViewTaskActivity activity, Step step);
```

Comment:
I am not sure about the MainThread annotation. The callback is used only in one place: by the ViewTaskActivity, after the animation that goes from one step to another has been completed. It is likely that this happens on the main thread, but there is no special reason why this MUST happen there. Or am I missing something?